components/governance/org.wso2.carbon.governance.taxonomy.ui/src/main/java/org/wso2/carbon/governance/taxonomy/ui/clients/TaxonomyManagementClient.java

```diff
@@ -0,0 +1,132 @@
+        endpointURL = backendServerURL + "TaxonomyServices";
+
+        try {
+            stub = new TaxonomyServicesStub(configContext, endpointURL);
+            ServiceClient client = stub._getServiceClient();
+            Options option = client.getOptions();
+            option.setManageSession(true);
+            option.setProperty(org.apache.axis2.transport.http.HTTPConstants.COOKIE_STRING, cookie);
+
+        } catch (AxisFault axisFault) {
+            String msg = "Failed to initiate taxonomy service client. " + axisFault.getMessage();
+            throw new TaxonomyServicesRegistryExceptionException(msg, axisFault);
+        }
+    }
+
+    /**
+     * This method will call addTaxonomy method in taxonomy OSGI service
+     *
+     * @param request User request
+     * @return boolean
+     * @throws RemoteException
+     * @throws TaxonomyServicesRegistryExceptionException
+     */
+    public boolean newTaxonomy(HttpServletRequest request)
+            throws RemoteException, TaxonomyServicesRegistryExceptionException {
+        return stub.addTaxonomy(request.getParameter("payload"));
+    }
+
+    /**
+     * This method will return content of the taxonomy file
+     *
+     * @param request user request which contains taxonomy file name
+     * @return String content
+     * @throws RemoteException
+     * @throws TaxonomyServicesRegistryExceptionException
+     */
+    public String getTaxonomy(HttpServletRequest request)
+            throws RemoteException, TaxonomyServicesRegistryExceptionException {
+        return stub.getTaxonomy(request.getParameter("taxonomyName"));
+    }
+
+    /**
+     * This method will call updateTaxonomy method in taxonomy OSGI service
+     *
+     * @param request user request
+     * @return boolean
+     * @throws RemoteException
+     * @throws TaxonomyServicesRegistryExceptionException
+     */
+    public boolean updateTaxonomy(HttpServletRequest request)
+            throws RemoteException, TaxonomyServicesRegistryExceptionException {
+        return stub.updateTaxonomy(request.getParameter("taxonomyName"), request.getParameter("payload"));
+    }
+
+    /**
+     * This method will call deleteTaxonomy method in taxonomy OSGI service
+     *
+     * @param request user request
+     * @return boolean
+     * @throws RemoteException
+     * @throws TaxonomyServicesRegistryExceptionException
+     */
+    public boolean deleteTaxonomy(HttpServletRequest request)
+            throws RemoteException, TaxonomyServicesRegistryExceptionException {
+        return stub.deleteTaxonomy(request.getParameter("taxonomyName"));
+    }
+
+    /**
+     * This method will return array of taxonomies which in the registry for a specific tenant
+     *
+     * @param request user request
+     * @return String array of taxonomy files
+     * @throws RemoteException
+     * @throws TaxonomyServicesRegistryExceptionException
+     */
+
+    public String[] getTaxonomyList(HttpServletRequest request)
```

Comment:
Thank you Thushara for mentioning it,
It has been fixed with https://github.com/wso2/carbon-governance/pull/253